sdk/client/src/stream/streaming_client.rs

```diff
@@ -215,15 +242,29 @@ impl StreamingClient {
                 return Ok(());
             }
         };
+
+        // If this is an unsubscription confirmation, drop the subscription
+        let msg_is_unsubscribe = msg.result.as_ref().map_or(false, |v| v.get("unsubscribe").is_some());
+        if msg_is_unsubscribe {
+            self.clear_subscription(&id).await;
```

Comment:
I think i'm still having some difficulty understanding the need for these clear_subscription calls, I thought this was handled in drop?